src/dashboard/src/components/administration/views_processing.py

```diff
@@ -47,26 +47,51 @@ def list(request):
 
 
 def edit(request, name=None):
-    if request.method == 'POST':
-        form = ProcessingConfigurationForm(request.POST)
-        if form.is_valid():
-            try:
-                form.save_config()
-            except Exception:
-                msg = 'Failed to save processing config.'
-                logger.exception(msg)
-                messages.error(request, msg)
-            else:
-                messages.info(request, 'Saved!')
-            return redirect('components.administration.views_processing.list')
-    else:
-        form = ProcessingConfigurationForm()
-        if name is not None:
-            try:
-                form.load_config(name)
-            except IOError:
-                raise Http404
-    return render(request, 'administration/processing_edit.html', {'form': form})
+
```

Comment:
Functionally this function is very similar but with more exhaustive error handling, e.g. internal server error page is avoided as much as possible and error messages are more explicit, also better logging. The code is also written to avoid nested blocks, i.e. returning early.